addons/udes_stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -575,3 +614,293 @@ def drop_off_picked(self, continue_batch, move_line_ids, location_barcode, resul
         #     self.close()
 
         return self
+        
+    def get_single_batch(self, user_id=None):
+        """
+        Search for a picking batch in progress for the specified user.
+        If no user is specified, the current user is considered.
+
+        Raise a ValidationError in case it cannot perform a search
+        or if multiple batches are found for the specified user.
+        """
+        PickingBatch = self.env["stock.picking.batch"]
+
+        user_id = self._check_user_id(user_id)
+        batches = PickingBatch.search(
+            [("user_id", "=", user_id), ("state", "=", "in_progress")]
+        )
+        batch = None
+
+        if batches:
+            if len(batches) > 1:
+                raise ValidationError(
+                    _("Found %d batches for the user, please contact " "administrator.")
+                    % len(batches)
+                )
+
+            batch = batches
+
+        return batch
+
+    def _check_user_id(self, user_id):
```

Comment:
Nit: This method might be better named as `_get_user_id` as it's not really checking much

Reply:
Not sure I understand, as it seems to me that it checks that there is a user_id input and will return the environment user if no input is given or raise an exception if it set to false.

Reply:
The way I read it is that it's returning a user_id therefore it's getting a user_id rather than checking. In reality you could argue that it's doing a bit of both.